### PR TITLE
kubelet: fix raw block mode CSI NodePublishVolume stage miss pod info

### PR DESCRIPTION
**Pull request by phantooom**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
fix raw block mode CSI NodePublishVolume stage miss pod info
#### Which issue(s) this PR fixes:
https://github.com/kubernetes/kubernetes/issues/98832
```release-note
fix raw block mode CSI NodePublishVolume stage miss pod info
```

### CODE REVIEW

**Comment by phantooom** [comment]
@Jiawei0227  I fixed test & rebase the commit. please rerun the test. thanks.

**Comment by msau42** [comment]
/assign  @chrishenzie 
@kubernetes/sig-storage-pr-reviews

**Comment by chrishenzie** [comment]
This all looks good to me. Do we have any unit test coverage for this change?

**Comment by phantooom** [comment]
> This all looks good to me. Do we have any unit test coverage for this change? (Mounting a pod with a CSI driver that specifies podInfoOnMount?)

we don't have ut cover this change.

**Comment by chrishenzie** [comment]
Can a unit test that covers this be added? (Or can an existing test be extended to cover this?)

It looks like the fake CSI volume that gets published has a field for the `VolumeContext` that would contain the pod info that was added in this PR
https://github.com/kubernetes/kubernetes/blob/b2860a360441d15a2f1a7b63e9ca247e5b0ec784/pkg/volume/csi/fake/fake_client.go#L73

And I think the `pvol` published volume that we get back contains this field that would let us assert the correct context was provided
https://github.com/kubernetes/kubernetes/blob/b2860a360441d15a2f1a7b63e9ca247e5b0ec784/pkg/volume/csi/csi_block_test.go#L332-L334

**Comment by ehashman** [comment]
/triage accepted
/priority important-longterm

Looks like this needs rebase.

**Comment by phantooom** [comment]
test failed will be fixed by https://github.com/kubernetes/minikube/pull/10537

**Comment by phantooom** [comment]
> Can a unit test that covers this be added? (Or can an existing test be extended to cover this?)
> 
> It looks like the fake CSI volume that gets published has a field for the `VolumeContext` that would contain the pod info that was added in this PR
> https://github.com/kubernetes/kubernetes/blob/b2860a360441d15a2f1a7b63e9ca247e5b0ec784/pkg/volume/csi/fake/fake_client.go#L73
> 
> And I think the `pvol` published volume that we get back contains this field that would let us assert the correct context was provided
> https://github.com/kubernetes/kubernetes/blob/b2860a360441d15a2f1a7b63e9ca247e5b0ec784/pkg/volume/csi/csi_block_test.go#L332-L334

@chrishenzie I add a ut cover this case.

**Comment by phantooom** [comment]
> /triage accepted
> /priority important-longterm
> 
> Looks like this needs rebase.

@ehashman done

**Comment by phantooom** [comment]
@msau42 @ehashman @chrishenzie 

What should i do next. Is there any suggestion?

**Comment by ehashman** [comment]
I've flagged your PR for review from Node. I don't know if we should consider this for test freeze; I suggest you reach out to either #pr-reviews or someone from SIG Storage.

**Comment by phantooom** [comment]
> I've flagged your PR for review from Node. I don't know if we should consider this for test freeze; I suggest you reach out to either #pr-reviews or someone from SIG Storage.

i see

**Comment by phantooom** [comment]
@chrishenzie
This pr new review thx～

File: pkg/volume/csi/csi_util.go [hunk 1]
```diff
@@ -203,3 +204,17 @@ func createCSIOperationContext(volumeSpec *volume.Spec, timeout time.Duration) (
 	ctx := context.WithValue(context.Background(), additionalInfoKey, additionalInfo{Migrated: strconv.FormatBool(migrated)})
 	return context.WithTimeout(ctx, timeout)
 }
+
+// GetPodInfoAttrs returns pod info for NodePublish
+func GetPodInfoAttrs(pod *api.Pod, volumeMode storage.VolumeLifecycleMode) map[string]string {
```
Comment (msau42) [inline pkg/volume/csi/csi_util.go:209]: Does this need to be public? It's within the same package.

**Review by msau42 — COMMENTED** [review]
I just have a minor comment, otherwise lgtm. Thanks for working on it!

/approve
/hold
to address comment

**Comment by ehashman** [comment]
/remove-sig node

No node files in scope, this is storage.

File: pkg/volume/csi/csi_util.go (hunk 1 above)
Comment (phantooom) [reply pkg/volume/csi/csi_util.go]: @msau42 resloved～

**Comment by phantooom** [comment]
@msau42 @chrishenzie  resloved

**Pull request merged**